mitiq/interface/conversions.py

```diff
@@ -248,8 +245,6 @@ def new_scaling_function(
             from pyquil import Program
             from pyquil.quilbase import Declare, Measurement
 
-            circuit = cast(Program, circuit)
-
```

Comment:
I'm getting circuits typed as any after this. I think it's good practice to do what we do in the `qiskit` pass above: `circuit = circuit.copy()`?

Reply:
I added the `cast` call back in [`3af0b0f`](https://github.com/unitaryfund/mitiq/pull/1825/commits/3af0b0f3e69a16053fa6539ca9d8e2d5a07906e9), but I think since we don't need to modify the program for a pyquil program, we do not need to copy the circuit as we do with qiskit.